JustSaying/Messaging/MessageHandling/IMessageContextReader.cs

```diff
@@ -0,0 +1,7 @@
+namespace JustSaying.Messaging.MessageHandling
+{
+    public interface IMessageContextReader
```

Comment:
Since it let's us read and write, this should be named something like `IMessageContextAccessor`.